<a href="https://colab.research.google.com/github/ksuaray/Suaray25/blob/main/STAT108_IntroStats_College_Majors_Assignment_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Guide to Picking a College Major**

#**Importing Necessary Python Modules**

Python incorporates a variety of open source add-ins called **modules** that enable us to be able to

In [ ]:
import pandas as pd
import numpy as np
import plotly.express as px
from IPython.display import Image


#**Context**

This assignment contains the data behind the story The Economic Guide To Picking A College Major. The data is from the American Community Survey 2010-2012 Public Use Microdata Series. With this dataset, you'll have the power to explore college programs and their graduates like never before and create stories of your own!

In [ ]:
# Replace 'image_url' with the URL of the image you want to display
image_url = 'https://media.istockphoto.com/id/1470208665/photo/multi-ethnic-group-of-latin-and-african-american-college-students-smiling-diversity-portrait.jpg?s=2048x2048&w=is&k=20&c=zicp2F74iFTRKjJUwFBgs_Mb_Xd5vvkvdmYSVoekb1I='

# Display the image
Image(url=image_url)

#**About the Dataset**

This dataset contains 172 rows corresponding to a random sample of people with at least some college. The variables provided are listed below:

**Variables**

| Column name                 | Description                                                |
|-----------------------------|------------------------------------------------------------|
| Index                       | A number assigned to an individual                         |
| Major_code                  | The code associated with the major (Integer)               |
| Major                       | The specific major of the field of study (String)          |
| Major_category              | The category of the major (String)                         |
| Grad_total                  | The total number of graduates from the major (Integer)     |
| Grad_sample_size            | The sample size of graduates from the major (Integer)      |
| Grad_employed               | The number of graduates employed (Integer)                 |
| Grad_full_time_year_round   | The number of graduates employed full-time year-round (Integer) |
| Grad_unemployed             | The number of graduates unemployed (Integer)               |
| Grad_unemployment_rate      | The unemployment rate of graduates (Float)                 |
| Grad_median                 | The median salary of graduates (Integer)                   |
| Grad_P25                    | The 25th percentile salary of graduates (Integer)          |
| Grad_P75                    | The 75th percentile salary of graduates (Integer)          |
| Nongrad_total               | The total number of non-graduates from the major (Integer) |
| Nongrad_employed            | The number of non-graduates employed (Integer)             |
| Nongrad_full_time_year_round| The number of non-graduates employed full-time year-round (Integer) |
| Nongrad_unemployed          | The number of non-graduates unemployed (Integer)           |
| Nongrad_unemployment_rate   | The unemployment rate of non-graduates (Float)             |
| Nongrad_median              | The median salary of non-graduates (Integer)               |
| Nongrad_P25                 | The 25th percentile salary of non-graduates (Integer)      |
| Nongrad_P75                 | The 75th percentile salary of non-graduates (Integer)      |
| Grad_share                  | The share of graduates in the major (Float)                |
| Grad_premium                | The difference between the median salary of graduates and non-graduates (Integer) |


*Attribution:  FiveThirtyEight.com*

We can view a snippet of the data by first importing it directly from the [url below](https://raw.githubusercontent.com/fivethirtyeight/data/master/college-majors/grad-students.csv).

**Data**

In [ ]:
file_path = "https://raw.githubusercontent.com/fivethirtyeight/data/master/college-majors/grad-students.csv"
df = pd.read_csv(file_path)


Next, we can display the data by typing the name of the DataFrame. To ensure we can see all columns, we'll use the *pd.set_option* method.

In [ ]:
# Set display options to show all columns
pd.set_option('display.max_columns', None)
df

Major_code                                       Major  \
0          5601                       CONSTRUCTION SERVICES   
1          6004           COMMERCIAL ART AND GRAPHIC DESIGN   
2          6211                      HOSPITALITY MANAGEMENT   
3          2201      COSMETOLOGY SERVICES AND CULINARY ARTS   
4          2001                  COMMUNICATION TECHNOLOGIES   
..          ...                                         ...   
168        5203                       COUNSELING PSYCHOLOGY   
169        5202                         CLINICAL PSYCHOLOGY   
170        6106     HEALTH AND MEDICAL PREPARATORY PROGRAMS   
171        2303                   SCHOOL STUDENT COUNSELING   
172        2301  EDUCATIONAL ADMINISTRATION AND SUPERVISION   

                          Major_category  Grad_total  Grad_sample_size  \
0    Industrial Arts & Consumer Services        9173               200   
1                                   Arts       53864               882   
2                               Business       24417               437   
3    Industrial Arts & Consumer Services        5411                72   
4                Computers & Mathematics        9109               171   
..                                   ...         ...               ...   
168             Psychology & Social Work       51812               724   
169             Psychology & Social Work       22716               355   
170                               Health      114971              1766   
171                            Education       19841               260   
172                            Education       54159               841   

     Grad_employed  Grad_full_time_year_round  Grad_unemployed  \
0             7098                       6511              681   
1            40492                      29553             2482   
2            18368                      14784             1465   
3             3590                       2701              316   
4             7512                       5622              466   
..             ...                        ...              ...   
168          38468                      28808             1420   
169          16612                      12022              782   
170          78132                      58825             1732   
171          11313                       8130              613   
172          34142                      26850              582   

     Grad_unemployment_rate  Grad_median  Grad_P25  Grad_P75  Nongrad_total  \
0                  0.087543      75000.0     53000  110000.0          86062   
1                  0.057756      60000.0     40000   89000.0         461977   
2                  0.073867      65000.0     45000  100000.0         179335   
3                  0.080901      47000.0     24500   85000.0          37575   
4                  0.058411      57000.0     40600   83700.0          53819   
..                      ...          ...       ...       ...            ...   
168                0.035600      50000.0     36000   65000.0          16781   
169                0.044958      70000.0     47000   95000.0           6519   
170                0.021687     135000.0     70000  294000.0          26320   
171                0.051400      56000.0     42000   70000.0           2232   
172                0.016761      65000.0     50000   86000.0           4003   

     Nongrad_employed  Nongrad_full_time_year_round  Nongrad_unemployed  \
0               73607                         62435                3928   
1              347166                        250596               25484   
2              145597                        113579                7409   
3               29738                         23249                1661   
4               43163                         34231                3389   
..                ...                           ...                 ...   
168             12377                          8502                 835   
169              

#**ASSIGNMENT 1 - Descriptive Statistics: Graphical and Numerical Summary**

**INSTRUCTIONS**

**Use Python to analyze the data set and complete each of the following. As appropriate, copy the SPSS output and paste it in the correct part below. For problems that require a written response, type the answer below.**

##**QUESTION 1**

**Determine whether the three variables below are qualitative or quantitative. If they are quantitative, specify whether they are continuous or discrete.**

| Variable         | Qual or Quant | Dis, Con, or Neither |
|------------------|---------------|----------------------|
| **Major**              | Qual or Quant  | Dis, Con, or Neither              |
| **Grad_unemployment_rate**           | Qual or Quant   | Dis, Con, or Neither              |
| **Grad_Median**           | Qual or Quant  | Dis, Con, or Neither           |


##**QUESTION 2**

**Construct a frequency table, relative frequency table, and relative frequency bar chart to describe the distribution of Major_category. State any fact that jumps out to you.**

In [ ]:
#Frequency table
freq_table = pd.value_counts(df['Major_category'])
freq_table

Engineering                            29
Education                              16
Humanities & Liberal Arts              15
Biology & Life Science                 14
Business                               13
Health                                 12
Computers & Mathematics                11
Agriculture & Natural Resources        10
Physical Sciences                      10
Social Science                          9
Psychology & Social Work                9
Arts                                    8
Industrial Arts & Consumer Services     7
Law & Public Policy                     5
Communications & Journalism             4
Interdisciplinary                       1
Name: Major_category, dtype: int64

In [ ]:
#Relative frequency table
freq_table/len(df)

Engineering                            0.167630
Education                              0.092486
Humanities & Liberal Arts              0.086705
Biology & Life Science                 0.080925
Business                               0.075145
Health                                 0.069364
Computers & Mathematics                0.063584
Agriculture & Natural Resources        0.057803
Physical Sciences                      0.057803
Social Science                         0.052023
Psychology & Social Work               0.052023
Arts                                   0.046243
Industrial Arts & Consumer Services    0.040462
Law & Public Policy                    0.028902
Communications & Journalism            0.023121
Interdisciplinary                      0.005780
Name: Major_category, dtype: float64

In [ ]:
dfrf = pd.DataFrame(freq_table)
fig = px.bar(x=dfrf.index, y=dfrf['Major_category'], barmode='group',
             title='Frequency Distribution Bar Chart')
fig.show()

## **QUESTIONS 3-6**

**For questions 3-6: Find your variable based on your last name and use that variable when answering questions #3 to #6.**  

| Last Name | Variable                  |
|-----------|---------------------------|
| A-F       | Grad_unemployment_rate    |
| G-M       | Grad_median               |
| N-S       | Nongrad_unemployment_rate |
| T-Z       | Nongrad_median            |


In [ ]:
#Choose variable here

variable = 'Grad_unemployment_rate'

###**QUESTION 3**

**Construct a histogram for your variable. Use Number of Intervals = 12.**

In [ ]:
fig = px.histogram(x=df[variable],nbins = 12)
fig.show()

###**QUESTION 4**

**Construct a boxplot for your variable.**  

In [ ]:
px.box(x=df[variable])

###**QUESTION 5**

**Calculate the following summary statistics for your variable: minimum, maximum, mean, median, standard deviation, Q1, and Q3. Paste the output below.**

In [ ]:
df[variable].describe()

count    173.000000
mean       0.039343
std        0.019076
min        0.000000
25%        0.026068
50%        0.036654
75%        0.048051
max        0.138515
Name: Grad_unemployment_rate, dtype: float64

###**QUESTION 6**

**Use information from questions #3, #4, and #5 to describe your variable in terms of shape, center, spread, and outliers. Interpret your findings.**

*The distribution of ______ is ______.*
*The mean/median is ______*
*The standard deviation/IQR is ______*
*And there are ___ outliers in the _____ tail.*

*The distribution of `Grad_unemployment_rate` is `skewed right`.*
*The `median` is `3.67%`*
*The `IQR` is `2.20%`*
*And there are `7` outliers in the `right` tail.*

##**QUESTION 7**

**Calculate and state the unemployment rate for graduates, the median salary for graduates, the unemployment rate for non-graduates, and the median salary for non-graduates for your major or intended major. Compare the results.**

In [ ]:
df1 = df[['Major','Grad_unemployment_rate','Grad_median','Nongrad_unemployment_rate','Nongrad_median']]
df1['Unemp_diff'] = df1['Nongrad_unemployment_rate'] - df1['Grad_unemployment_rate']
df1['Sal_diff'] = df1['Grad_median'] - df1['Nongrad_median']
df1

<ipython-input-33-2ee7cafd3ace>:2: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy

<ipython-input-33-2ee7cafd3ace>:3: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



Major  Grad_unemployment_rate  \
0                         CONSTRUCTION SERVICES                0.087543   
1             COMMERCIAL ART AND GRAPHIC DESIGN                0.057756   
2                        HOSPITALITY MANAGEMENT                0.073867   
3        COSMETOLOGY SERVICES AND CULINARY ARTS                0.080901   
4                    COMMUNICATION TECHNOLOGIES                0.058411   
..                                          ...                     ...   
168                       COUNSELING PSYCHOLOGY                0.035600   
169                         CLINICAL PSYCHOLOGY                0.044958   
170     HEALTH AND MEDICAL PREPARATORY PROGRAMS                0.021687   
171                   SCHOOL STUDENT COUNSELING                0.051400   
172  EDUCATIONAL ADMINISTRATION AND SUPERVISION                0.016761   

     Grad_median  Nongrad_unemployment_rate  Nongrad_median  Unemp_diff  \
0        75000.0                   0.050661         65000.0   -0.036882   
1        60000.0                   0.068386         48000.0    0.010630   
2        65000.0                   0.048423         50000.0   -0.025444   
3        47000.0                   0.052900         41600.0   -0.028001   
4        57000.0                   0.072800         52000.0    0.014390   
..           ...                        ...             ...         ...   
168      50000.0                   0.063200         40000.0    0.027600   
169      70000.0                   0.075556         46000.0    0.030598   
170     135000.0                   0.058725         51000.0    0.037038   
171      56000.0                   0.112892         42000.0    0.061492   
172      65000.0                   0.000000         58000.0   -0.016761   

     Sal_diff  
0     10000.0  
1     12000.0  
2     15000.0  
3      5400.0  
4      5000.0  
..        ...  
168   10000.0  
169   24000.0  
170   84000.0  
171   14000.0  
172    7000.0  

[173 rows x 7 columns]

##**QUESTION 8**

**Generate a paragraph of at least 100 words to address one of the following questions:**

### **QUESTION 8a**

**Discuss how analyzing your chosen data set using statistical methods could help you become better prepared for future courses in your major?**

### **QUESTION 8b**

**Discuss how analyzing your chosen data set using statistical methods could be instrumental in becoming better prepared for your future career?**